src/main/java/com/kodilla/patterns2/adapter/bookclasifier/libraryb/Statistics.java

```diff
@@ -0,0 +1,32 @@
+package com.kodilla.patterns2.adapter.bookclasifier.libraryb;
+
+import java.util.Arrays;
+import java.util.Map;
+
+public class Statistics implements BookStatistics{
+    @Override
+    public int averagePublicationYear(Map<BookSignature, Book> books){
+        if (books.size() == 0)return 0;
+        int sum = 0;
+        for (Map.Entry<BookSignature, Book> entry:books.entrySet()){
+            sum += entry.getValue().getYearOfPublication();
+        }
+        return sum / books.size();
+    }
+    @Override
+    public int medianPublicationYear(Map<BookSignature, Book> books){
+        if (books.size()==0)return 0;
+        int[]years = new int[books.size()];
+        int n = 0;
+        for (Map.Entry<BookSignature, Book> entry:books.entrySet()){
+            years[n] = entry.getValue().getYearOfPublication();
+            n++;
+        }
+        Arrays.sort(years);
+        if (years.length % 2 == 0){
```

Comment:
nie tak się wylicza mediane :) mediana przy parzystej liczbie elementów to suma dwóch środkowych elementów podzielona przez 2